### PR TITLE
Links between companions after unpickling

### PR DESCRIPTION
**Pull request by VladimirNik**
Fix for the problem with missing links between companions during unpickling:
https://github.com/lampepfl/dotty/issues/881

Review by @DarkDimius @odersky

### CODE REVIEW

File: src/dotty/tools/dotc/core/tasty/TreeUnpickler.scala [hunk 1]
```diff
@@ -588,9 +588,19 @@ class TreeUnpickler(reader: TastyReader, tastyName: TastyName.Table) {
           sym.info = readType()
           ValDef(sym.asTerm, readRhs(localCtx))
         case TYPEDEF | TYPEPARAM =>
-          if (sym.isClass)
+          def addCompanionMethod(target: Symbol, owner: Symbol) = {
```
Comment (DarkDimius) [inline src/dotty/tools/dotc/core/tasty/TreeUnpickler.scala]: This method is equivalent to `registerCompanionPair` in scala2Unpickler. @VladimirNik, would you consider removing code duplication?

**Comment by DarkDimius** [comment]
Otherwise LGTM.

**Comment by VladimirNik** [comment]
@DarkDimius these methods are not equivalent. It seems that `registerCompanionPair` in scala2Unpickler duplicates companion methods:
```scala
module.unforcedDecls: Scopes(val FLAG_SERIALIZABLE, val FLAG_MARKERS, val FLAG_BRIDGES, method metafactory, method altMetafactory, method <clinit>, method companion$class, method companion$class, method companion$class)
claz.unforcedDecls: Scopes(method <init>, method companion$module, method companion$module, method companion$module)
```
Maybe it's better to add fixes for this issue in the separate pull request?

**Comment by VladimirNik** [comment]
This pull request should be rebased on https://github.com/lampepfl/dotty/pull/956/files

**Pull request closed without merging**